qiskit/circuit/parameterexpression.py

```diff
@@ -156,36 +167,38 @@ def subs(self, parameter_map: Dict) -> "ParameterExpression":
         Returns:
             A new expression with the specified parameters replaced.
         """
-        inbound_parameters = set()
-        inbound_names = {}
-        for replacement_expr in parameter_map.values():
-            for p in replacement_expr.parameters:
-                inbound_parameters.add(p)
-                inbound_names[p.name] = p
-
-        self._raise_if_passed_unknown_parameters(parameter_map.keys())
+        if not allow_unknown_parameters:
+            self._raise_if_passed_unknown_parameters(parameter_map.keys())
+
+        inbound_names = {
+            p.name: p
+            for replacement_expr in parameter_map.values()
+            for p in replacement_expr.parameters
+        }
         self._raise_if_parameter_names_conflict(inbound_names, parameter_map.keys())
+
+        # Include existing parameters in self not set to be replaced.
+        new_parameter_symbols = {
+            p: s for p, s in self._parameter_symbols.items() if p not in parameter_map
```

Comment:
It would be nice to be specific here, I'm always wondering what `in <dict>` actually checks again -- but if that's just me feel free to leave as is 🙂 
```suggestion
            p: s for p, s in self._parameter_symbols.items() if p not in parameter_map.keys()
```